evalml/data_checks/class_imbalance_data_check.py

```diff
@@ -28,17 +33,25 @@ def validate(self, X, y):
             y: Target labels to check for imbalanced data.
 
         Returns:
-            list (DataCheckWarning): list with DataCheckWarnings if imbalance in classes is less than the threshold.
+            list (DataCheckWarning, DataCheckError): list with DataCheckWarnings if imbalance in classes is less than the threshold,
+                                                     and DataCheckErrors if the number of values for each target is below 2 * num_cv_folds.
 
         Example:
             >>> X = pd.DataFrame({})
             >>> y = pd.Series([0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
             >>> target_check = ClassImbalanceDataCheck(threshold=0.10)
-            >>> assert target_check.validate(X, y) == [DataCheckWarning("The following labels fall below 10% of the target: [0]", "ClassImbalanceDataCheck")]
+            >>> assert target_check.validate(X, y) == [DataCheckError("The number of instances of these targets is less than 2 * the number of cross folds = 6 instances: [0]", "ClassImbalanceDataCheck"), DataCheckWarning("The following labels fall below 10% of the target: [0]", "ClassImbalanceDataCheck")]
         """
         if not isinstance(y, pd.Series):
             y = pd.Series(y)
         messages = []
+        fold_counts = y.value_counts(normalize=False)
+        # search for targets that occur less than twice the number of cv folds first
+        below_threshold_folds = fold_counts.where(fold_counts < self.cv_folds).dropna()
+        if len(below_threshold_folds):
+            warning_msg = "The number of instances of these targets is less than 2 * the number of cross folds = {} instances: {}"
```

Comment:
Helpful error message! But it might be confusing if the user passes in a custom 4 fold CV and see `2 * number of cross folds = 6` in the message. Is there a plan to modify AutoMLSearch to init this class with the number of folds in the cv? Or maybe we make this more generic like "The number of instances of these targets is too small for cross validation? Just thinking out loud - this is fine to merge and we can refine if we get user feedback 😅

Reply:
@freddyaboulton Mmmm that's fair, I didn't think about that. I think it would make sense to modify such that we can pass `num_folds` arg from AutoMLSearch to the ClassImbalanceDataCheck itself. For instance, if the user passed in a custom data-splitter with > 6 folds, this data check now does nothing, and if they use the default data checks, it could pass that while failing to catch the error. I think that can be filed as a separate issue, unless it's better to implement it into this PR?

Reply:
@freddyaboulton filed that as an issue [here](https://github.com/alteryx/evalml/issues/1345). I'll merge this PR and we can address the `num_folds` arg in AutoMLSearch afterwards

Reply:
Good point @freddyaboulton and thank you for filing @bchen1116 ! Yep sounds good. I agree this message can be improved.